app/models/aspace/resource.rb

```diff
@@ -1,23 +1,40 @@
 # frozen_string_literal: true
 
 module Aspace
+  class AspaceResourceError < StandardError; end
+
   # Model for an Aspace resource to manage information
   # needed for downloading and updating resources in ArcLight
   class Resource
-    attr_reader :ead_id, :uri, :repository_code
+    attr_reader :ead_id, :uri, :identifier, :repository_code
 
-    def initialize(ead_id:, uri:, repository_code:)
+    def initialize(uri:, repository_code:, ead_id: nil, identifier: nil)
       @ead_id = ead_id
+      @identifier = identifier
       @uri = uri
       @repository_code = repository_code
     end
```

Comment:
I'd prefer if the parameters/variables were listed in the same order each time so it's easier to understand at a glance that they're the same.